libnavigation-base/src/main/java/com/mapbox/navigation/base/trip/model/roadobject/custom/Custom.kt

```diff
@@ -0,0 +1,62 @@
+package com.mapbox.navigation.base.trip.model.roadobject.custom
+
+import com.mapbox.navigation.base.trip.model.roadobject.RoadObject
+import com.mapbox.navigation.base.trip.model.roadobject.RoadObjectGeometry
+import com.mapbox.navigation.base.trip.model.roadobject.RoadObjectType
+
+/**
+ * Road object type that provides information about custom objects.
+ *
+ * @see RoadObject
+ * @see RoadObjectType.Custom
+ */
+class Custom private constructor(
+    distanceFromStartOfRoute: Double?,
+    objectGeometry: RoadObjectGeometry
+) : RoadObject(
+    RoadObjectType.Custom,
+    distanceFromStartOfRoute,
+    objectGeometry
+) {
+
+    /**
+     * Transform this object into a builder to mutate the values.
+     */
+    fun toBuilder(): Builder = Builder(
+        objectGeometry
+    )
+        .distanceFromStartOfRoute(distanceFromStartOfRoute)
+
+    /**
+     * Returns a string representation of the object.
+     */
+    override fun toString(): String {
+        return "Custom() ${super.toString()}"
+    }
+
+    /**
+     * Use to create a new instance.
+     *
+     * @see Custom
+     */
+    class Builder(
+        private val objectGeometry: RoadObjectGeometry
+    ) {
+        private var distanceFromStartOfRoute: Double? = null
+
+        /**
+         * Add optional distance from start of route.
+         */
+        fun distanceFromStartOfRoute(distanceFromStartOfRoute: Double?): Builder = apply {
```

Comment:
the same https://github.com/mapbox/mapbox-navigation-android/pull/4109/files#r592430961

Reply:
discussed above